driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -115,13 +123,17 @@ public InternalConnection get(final long timeout, final TimeUnit timeUnit) {
                         pooledConnection.open();
                     } catch (Throwable t) {
                         pool.release(pooledConnection.wrapped, true);
+                        connectionPoolListener.connectionCheckOutFailed(new ConnectionCheckOutFailedEvent(serverId,
+                                Reason.CONNECTION_ERROR));
                         if (t instanceof MongoException) {
                             throw (MongoException) t;
                         } else {
                             throw new MongoInternalException(t.toString(), t);
                         }
                     }
                 }
+                connectionPoolListener.connectionCheckedOut(
```

Comment:
Seems like the getAsync method would need the same changes as the get method.  Are we testing the async methods at all in the test runner?